### PR TITLE
277 add notification when there are not enough stars

### PR DESCRIPTION
**Pull request by omen273**

### CODE REVIEW

File: app/src/androidTest/java/com/omen273/crossLingo/Utils.kt [hunk 1]
```diff
@@ -85,7 +85,9 @@ fun getItemFromCrosswordList(row: Int, column: Int, name: String? = null): Match
     return nthChildOf(nthChildOf(nthChildOf(withId(R.id.tableLayout), row), column), 0)
 }
 
-fun generateCrossword(pressHome: Boolean = true, chooseTopics: List<Int> = listOf(0)): Crossword {
+enum class FinishTypeGenerateCrossword {PRESS_HOME, PRESS_BACK, NOTHING}
+fun generateCrossword(finish: FinishTypeGenerateCrossword  = FinishTypeGenerateCrossword.PRESS_BACK,
```
Comment (greenEkatherine) [inline app/src/androidTest/java/com/omen273/crossLingo/Utils.kt:89]: remove extra space

File: app/src/main/res/values-ru-rRU/strings.xml [hunk 2]
```diff
@@ -27,6 +27,10 @@
     <string name="move_selection_to_solved_squares">печатать в решенные клетки</string>
     <string name="error_reasons_string">Пожалуйста, выберите тип ошибки</string>
     <string name="error_message">Пожалуйста, опишите ошибку ниже</string>
+    <string name="not_enough_stars_word">У вас недостаточно звезд (%1s\u2605) для открытия слова (-3\u2605).
+        Пожалуйста, решите кроссвод, чтобы получить звезды (+5\u2605)</string>
+    <string name="not_enough_stars_square">У вас недостаточно звезд(%1s\u2605) для открытия клетки (-1\u2605)
+        . Пожалуйста, решите кроссвод, чтобы получить звезды (+5\u2605)</string>
```
Comment (greenEkatherine) [inline app/src/main/res/values-ru-rRU/strings.xml:33]: почему точка на следующей строке

Comment (omen273) [reply app/src/main/res/values-ru-rRU/strings.xml:33]: Это не учитывается в выводе. Ты проверяла.

File: app/src/androidTest/java/com/omen273/crossLingo/Utils.kt [hunk 3]
```diff
@@ -94,10 +96,11 @@ fun generateCrossword(pressHome: Boolean = true, chooseTopics: List<Int> = listO
     onView(isRoot()).perform(waitForView(withId(R.id.ok_play)))
     onView(withId(R.id.ok_play)).perform(ViewActions.click())
     onView(isRoot()).perform(waitForView(withId(R.id.crossword)))
-    if (!pressHome) {
-        Espresso.pressBack()
-    } else {
+    when(finish) {
+        FinishTypeGenerateCrossword.PRESS_BACK -> Espresso.pressBack()
```
Comment (greenEkatherine) [inline app/src/androidTest/java/com/omen273/crossLingo/Utils.kt:100]: what's the difference between back and home?

File: app/src/main/java/com/omen273/crossLingo/GameActivity.kt [hunk 4]
```diff
@@ -424,14 +445,16 @@ class GameActivity : AppCompatActivity(), CrosswordView.OnLongPressListener,
...
+    private fun showFinishGameDialog(shownAgain: Boolean = false) {
+        val dialog = AlertDialog.Builder(this)
             .setPositiveButton(R.string.reset) { _, _ -> crosswordView.reset() }
             .setNegativeButton(R.string.another_crossword) { _, _ -> onBackPressed() }
             .setNeutralButton(R.string.remove) { _, _ ->
                 delete = true
                 onBackPressed()
-            }.create()
+            }
+        if (!shownAgain) dialog.setMessage(R.string.youve_solved_the_puzzle)
```
Comment (greenEkatherine) [inline app/src/main/java/com/omen273/crossLingo/GameActivity.kt:456]: why do you need this flag?

File: app/src/androidTest/java/com/omen273/crossLingo/Utils.kt (hunk 3 above)
Comment (omen273) [reply app/src/androidTest/java/com/omen273/crossLingo/Utils.kt:100]: Back is back and it is arrow on a right bottom angle home is home and it is arrow on a top left angle.

File: app/src/main/java/com/omen273/crossLingo/GameActivity.kt (hunk 4 above)
Comment (omen273) [reply app/src/main/java/com/omen273/crossLingo/GameActivity.kt:456]: We need it not to show the message R.string.youve_solved_the_puzzle when a user open the crossword again.

**Review by greenEkatherine — APPROVED** [review]

**Pull request merged**